dsc_lib/src/functions/mod.rs

```diff
@@ -50,6 +51,7 @@ impl FunctionDispatcher {
         let mut functions: HashMap<String, Box<dyn Function>> = HashMap::new();
```

Comment:
Can we have one end-to-end Pester test for each of these functions?
That would be also a good example demonstrating their use.

Reply:
We need what-if support as a way to execute the functions and return the results.  I could stick in `concat()` somewhere right now although it would be artificial.